### ISSUE TITLE
[Question] Getting Recomendations with Case_Study (already rated items in forecast)

### ISSUE THREAD

**Issue by sergey-semenov-ai** [RUCAIBox/RecBole#1131]
**Describe the bug**
Hello!
Model - BPR
DS - ml-100k
Using recbole.utils.case_study.full_sort_topk i get in forecast movies, that are already rated

**To Reproduce**

from recbole.quick_start import run_recbole, load_data_and_model
from recbole.utils.case_study import full_sort_topk
parameter_dict = {
   #'neg_sampling': None,
   'data_path': '/home/sergey/FL/Rollo/thinIce/',
   'dataset': 'ml-rb',
   'field_separator': ',',
   'load_col':  {'inter': ['user_id', 'item_id', 'rating']},
   'train_batch_size': 1024,
   'eval_batch_size': 1024   
}
run_recbole(model='BPR', config_dict=parameter_dict)

config, model, dataset, train_data, valid_data, test_data = load_data_and_model(
    model_file='saved/BPR-Jan-26-2022_11-01-45.pth')
uid_series = dataset.token2id(dataset.uid_field, ['611'])

topk_score, topk_iid_list = full_sort_topk(uid_series, model, test_data, k=10, device=config['device'])
external_item_list = dataset.id2token(dataset.iid_field, topk_iid_list.cpu())
print(external_item_list)  

**Expected behavior**
As I understand recbole.utils.case_study.full_sort_topk returns recommendations for movies that the user has not yet rated.
I am doing the following:
I manually add 2 scores for user 611 to the intersection file.
Next I train the model and receive recommendations for user 611. In these recommendations, I see one movie that I manually rated (with a rating of 0).
Next, I add all the recommendations received to the intersection file for user 611.
I reload the model with data and get the forecast again.
The new forecast contains 1-2 films that were also ranked at the previous step.
Why do rated films get recommended again?

**Screenshots**

![Снимок экрана от 2022-01-26 13-10-36](https://user-images.githubusercontent.com/70337257/151126451-539187c7-4d8e-46a9-93c1-50eb8d6f4dba.png)

**Desktop (please complete the following information):**
 - OS: ubuntu 18.04 LTS
- RecBole Version 1.0.0
 - Python Version 3.8.8
- PyTorch Version 1.10.1+cu102
- cudatoolkit Version  none

**Comment by Sherry-XLL** [comment]
@sergey-semenov-ai If you use the default configuration, RecBole will divide the training set, validation set and test set according to the ratio of `8:1:1`. The interaction you added may not be divided into the training set, so the items that the user interacted with in the inter file may appear in the recommendation list. Our purpose is to train with the interaction in the training set, in order that the model's recommendation of non-interactive items can be close to the interaction in the test set.

**Comment by sergey-semenov-ai** [comment]
Thanks!
I'm sorry I didn't figure it out myself.
Thanks again!
